crates/pipeline-manager/src/compiler.rs

```diff
@@ -566,12 +576,24 @@ inherits = "release"
                         e
                     );
                 }
-                db.transit_program_status_to_pending(
-                    tenant_id,
-                    pipeline.id,
-                    pipeline.program_version,
-                )
-                .await?;
+                match db
+                    .transit_program_status_to_pending(
+                        tenant_id,
+                        pipeline.id,
+                        pipeline.program_version,
+                    )
+                    .await
+                {
+                    Ok(()) => {}
+                    Err(e) => match e {
+                        DBError::UnknownPipeline { .. } => {
+                            info!("Program {} cannot be re-queued because it no longer exists (tenant: {tenant_id})", pipeline.id);
```

Comment:
```suggestion
                            debug!("Program {} cannot be re-queued because it no longer exists (tenant: {tenant_id})", pipeline.id);
```